src/utils/reserves.ts

```diff
@@ -0,0 +1,143 @@
+    const filters: GetProgramAccountsFilter[] = [
+        { dataSize: RESERVE_LEN },
+        { memcmp: { offset: 10, bytes: lendingMarketPubkey.toBase58() } },
+    ];
+    const reserves = await connection.getProgramAccounts(programId, {
+        commitment: connection.commitment,
+        filters,
+        encoding: "base64",
+    });
+    return reserves;
+};
+
+
+const getParsedReserve = (reserve: { pubkey: PublicKey; account: AccountInfo<Buffer> }) => {
+    const ParsedReserve = parseReserve(reserve.pubkey, reserve.account);
+    if (!ParsedReserve) {
+        throw new Error("Failed to parse reserve");
+    }
+    return ParsedReserve;
+};
+
+
+const getReserveViewModel = (
+    parsedReserve: ParsedReserve,
+    tokens: Map<string, TokenInfo>,
+    oraclePrices: Map<string, number>) => {
+
+    const { pubkey, info } = parsedReserve;
+    const tokenPubkey = info.liquidity.mintPubkey.toBase58();
+    const [supplyAPR, borrowAPR] = [calculateSupplyAPR(info), calculateBorrowAPR(info)];
+    const [supplyAPY, borrowAPY] = [calculateAPY(supplyAPR), calculateAPY(borrowAPR)];
+
+    let tokenInfo = tokens.get(tokenPubkey);
+    if (!tokenInfo) {
+        tokenInfo = {
+            tokenSymbol: `${tokenPubkey.slice(0, 4)}...${tokenPubkey.slice(-4)}`,
+            logoUri: null
+        } as TokenInfo;
+    }
+
+    const reserveViewModel = {
+        address: pubkey.toBase58(),
+        tokenSymbol: tokenInfo.tokenSymbol,
+        logoUri: tokenInfo.logoUri,
+        assetPriceUSD: oraclePrices.get(pubkey.toBase58()) || 0,
+        totalSupply: calculateTotalSuppliedAmount(info),
+        totalBorrow: calculateTotalBorrowedAmount(info),
+        LTV: calculateLoanToValueRatio(info),
+        supplyAPY: supplyAPY,
+        borrowAPY: borrowAPY,
+        supplyAPR: supplyAPR,
+        borrowAPR: borrowAPR,
+    };
+    return reserveViewModel;
+};
+
+
+const calculateTotalSuppliedAmount = (reserve: Reserve) => {
+    const supplyAmountWads = calculateSupplyAmountWads(reserve);
+    const decimals = BigNumber(reserve.liquidity.mintDecimals.toString());
+    const totalSupply = supplyAmountWads.dividedBy(BigNumber(10).pow(decimals.plus(BigNumber(18))));
+    return totalSupply;
+};
+
+
+const calculateTotalBorrowedAmount = (reserve: Reserve) => {
+    const borrowedAmountWads = BigNumber(reserve.liquidity.borrowedAmountWads.toString());
+    const decimals = BigNumber(reserve.liquidity.mintDecimals.toString());
+    const totalBorrow = borrowedAmountWads.dividedBy(BigNumber(10).pow(decimals.plus(BigNumber(18))));
+    return totalBorrow;
+};
+
+
+const calculateLoanToValueRatio = (reserve: Reserve) => {
+    const loanToValueRatio = reserve.config.loanToValueRatio;
+    return loanToValueRatio;
+};
+
+
+const reorderMainPoolReserves = (reserves: { pubkey: PublicKey; account: AccountInfo<Buffer> }[]) => {
+    const reservesMap = new Map<string, { pubkey: PublicKey; account: AccountInfo<Buffer> }>();
```

Comment:
I find using `.reduce` for this kind of stuff makes it a lot more readable. Getting used to using that can be nice. For example, this chunk could have been written as something like.:


```
const reservesMap = reserves.reduce((acc, pool) => ({
..acc,
[pool.pubkey.toBase58()]: pool,
});

const reorderedReserves = [
      ...MAIN_POOL_RESERVES_ADDRESSES.reduce(
        (acc, address) => {
          const prioritizedReserve = reservesMap[address];
          if (prioritizedReserve) {
            return [...acc, prioritizedReserve];
          }
          delete reservesMap[address]
          return acc;
        },
      ),
      ...reserves,
    ]```